zeek-ocsf/package.yaml

```diff
@@ -17,526 +17,585 @@ pipelines:
-          protocol_ver_id: protocol_ver_id,
-          protocol_name: event.proto,
-          protocol_num: protocol_num,
-        },
-        src_endpoint: {
-          ip: event.id.orig_h,
-          port: event.id.orig_p,
-          mac: event.orig_l2_addr,
-          location: {
-            country: event.orig_cc,
-          }
-        },
-        // TODO: we actually could go deeper into the `conn_state` field and
-        // choose a more accurate status. But this would require string
-        // manipulations and/or regex matching, which TQL doesn't have yet.
-        status: "Other",
-        status_code: event.conn_state,
-        status_id: 99,
-        traffic: {
-          bytes_in: event.resp_bytes,
-          bytes_out: event.orig_bytes,
-          packets_in: event.resp_pkts,
-          packets_out: event.orig_pkts,
-          total_bytes: event.orig_bytes + event.resp_bytes,
-          total_packets: event.orig_pkts + event.resp_pkts,
-        },
-        // --- Primary (optional) ---
-        // TODO
-        // - `ja4_fingerprint_list`: once we have some sample logs with JA4
-        //   fingerprints, which requires an additional Zeek package, we should
-        //   populate them here.
-        // - `tls`: if we buffer ssl log for this connection, we could add the
-        //   information in here.
-      }
-      // Drop all mapped fields.
-      drop(
-        unmapped._path,
-        unmapped._write_ts,
-        unmapped.community_id,
-        unmapped.conn_state,
-        unmapped.duration,
-        unmapped.id,
-        unmapped.local_orig,
-        unmapped.local_resp,
-        unmapped.orig_bytes,
-        unmapped.orig_cc,
-        unmapped.orig_l2_addr,
-        unmapped.orig_pkts,
-        unmapped.proto,
-        unmapped.resp_bytes,
-        unmapped.resp_cc,
-        unmapped.resp_l2_addr,
-        unmapped.resp_pkts,
-        unmapped.service,
-        unmapped.ts,
-        unmapped.uid,
+        ocsf.connection_info.direction = "Unknown"
+        ocsf.connection_info.direction_id = 0
+      }
+      drop (
+        zeek.community_id,
+        zeek.proto,
+        zeek.local_orig,
+        zeek.local_resp,
+      )
+      ocsf.status = "Other"
+      ocsf.status_code = zeek.conn_state
+      drop zeek.conn_state
+      ocsf.status_id = 99
+      ocsf.traffic = {
+        bytes_in: zeek.resp_bytes,
+        bytes_out: zeek.orig_bytes,
+        packets_in: zeek.resp_pkts,
+        packets_out: zeek.orig_pkts,
+        total_bytes: zeek.orig_bytes + zeek.resp_bytes,
+        total_packets: zeek.orig_pkts + zeek.resp_pkts,
+      }
+      drop (
+        zeek.resp_bytes,
+        zeek.orig_bytes,
+        zeek.resp_pkts,
+        zeek.orig_pkts,
       )
+      this = {...ocsf, unmapped: zeek}
       @name = "ocsf.network_activity"
       publish "ocsf"
 
   map-dns-to-ocsf:
     name: dns.log to OCSF DNS Activity
     description: Maps Zeek DNS logs to OCSF DNS Activity.
     definition: |
-      // tql2
```

Comment:
You removed the `// tql2` in most places (not all). I think it makes sense to still keep it, right?

Reply:
Because you'd need a TQL2-only node for it to work?

Reply:
Exactly.